server/src/main/java/io/druid/server/coordination/DruidServerMetadata.java

```diff
@@ -40,7 +43,10 @@ public DruidServerMetadata(
       @JsonProperty("maxSize") long maxSize,
       @JsonProperty("type") String type,
       @JsonProperty("tier") String tier,
-      @JsonProperty("priority") int priority
+      @JsonProperty("priority") int priority,
```

Comment:
@drcrallen From looking at the way DruidServerMetadata was created https://github.com/druid-io/druid/blob/master/server/src/main/java/io/druid/guice/StorageNodeModule.java#L62, I think it is intended to gather all the metadata information and serve as the source of true for server's metadata.